docs/2-building-blocks-of-oop-part-1/hometask/src/Point.ts

```diff
@@ -1 +1,46 @@
-export class Point {}
+export interface IPoint {
+  toString: () => string;
+  distance: (...args: any[]) => number;
+}
+
+export class Point implements IPoint {
+  readonly x: number;
+  readonly y: number;
+
+  public constructor();
+  public constructor(x: number, y: number);
+  public constructor(...args: any[]) {
+    if (args.length === 0) {
```

Comment:
in this case, it is better to use a comparison for the number of arguments, since the comparison for the number of arguments is also used below in the code. This increases the readability of the code and keeps it consistent.

Reply:
Are you suggesting me type a constructor?
<img width="458" alt="image" src="https://user-images.githubusercontent.com/50149163/178255682-34cf79be-7c53-4f6d-8306-0497b047ceb5.png">

Reply:
> in this case, it is better to use a comparison for the number of arguments, since the comparison for the number of arguments is also used below in the code. This increases the readability of the code and keeps it consistent.

ok, up to you

Reply:
> Are you suggesting me type a constructor? <img alt="image" width="458" src="https://user-images.githubusercontent.com/50149163/178255682-34cf79be-7c53-4f6d-8306-0497b047ceb5.png">

Firstly, constructor can't have return value. I suggest you if it is possible don't use any type at all, and if it is possible, to remove line 12

Reply:
Could you show me in code how it has to work? I don't understand your way to using overloading

if I remove line 12 I'll break the code
<img width="346" alt="image" src="https://user-images.githubusercontent.com/50149163/178337374-17016b3e-aa4a-4db6-a05a-a3ef7de1ef3c.png">

<img width="355" alt="image" src="https://user-images.githubusercontent.com/50149163/178337546-d46b3269-fea8-4563-913c-6eb87925d6bc.png">

Here it is really better to replace the `array of any` with an `array of numbers`, but completely remove the line 12th - I don’t understand what exactly is expected of me here.